src/main/java/io/github/opencubicchunks/cubicchunks/core/server/VanillaNetworkHandler.java

```diff
@@ -275,34 +344,48 @@ private void switchPlayerOffset(PlayerCubeMap cubeMap, EntityPlayerMP player, in
             }
         }
 
-        sendCubeLoadPackets(firstSendCubes, player);
+        Map<ChunkPos, ChunkPos> clientLoadedChunks = clientLoadedChunkPositions.computeIfAbsent(player, p -> new Object2ObjectOpenHashMap<>());
```

Comment:
Why is keeping track of clientLoadedChunkPositions separately necessary? PlayerCubeMap should have all the information necessary already (ColumnWatcher).

And since you are doing a horizontal switch, there may be columns sent to the client with no cubes sent yet. You should include them in `firstSendCubes` collection to avoid client memory leak (or more likely, geyser proxy in case of bedrock clients).

Reply:
Yeah, on second thought this was definitely overthought. See https://github.com/OpenCubicChunks/CubicChunks/pull/600#discussion_r507630137